fair/forward.py

```diff
@@ -1,326 +1,309 @@
-# Import required packages
+import inspect
 import numpy as np
 from scipy.optimize import root
+from constants import molwt, lifetime, radeff
+from constants.general import M_ATMOS
+from forcing import ozone_tr, ozone_st, h2o_st, contrails, aerosols, bc_snow,\
+                    landuse
 
-# Define a function which gives the relationship between iIRF_100 and scaling factor, alpha
-def iirf100_interp_funct(alpha,a,tau,targ_iirf100):
-    iirf100_arr = alpha*(np.sum(a*tau*(1.0 - np.exp(-100.0/(tau*alpha)))))
-    return iirf100_arr   -  targ_iirf100
+def iirf_interp_funct(alp_b,a,tau,targ_iirf):
+	# ref eq. (7) of Millar et al ACP (2017)
+    iirf_arr = alp_b*(np.sum(a*tau*(1.0 - np.exp(-100.0/(tau*alp_b)))))
+    return iirf_arr   -  targ_iirf
 
-# Define the FAIR simple climate model function
-def fair_scm(tstep=1.0,
-             emissions=False,
+def fair_scm(emissions=False,
              other_rf=0.0,
-             co2_concs=False,
              q=np.array([0.33,0.41]),
              tcrecs=np.array([1.6,2.75]),
              d=np.array([239.0,4.1]),
              a=np.array([0.2173,0.2240,0.2824,0.2763]),
              tau=np.array([1000000,394.4,36.54,4.304]),
              r0=32.40,
-             rC=0.019,
-             rT=4.165,
-             F_2x=3.74,
-             C_0=278.0,
-             ppm_gtc=2.123,
-             iirf100_max=97.0,
-             in_state=[[0.0,0.0,0.0,0.0],[0.0,0.0],0.0],
-             restart_out=False):
-
-  # # # ------------ CALCULATE Q ARRAY ------------ # # #
-  # If TCR and ECS are supplied, overwrite the q array
-  k = 1.0 - (d/70.0)*(1.0 - np.exp(-70.0/d))
+             rc=0.019,
+             rt=4.165,
+             F2x=3.74,
+             C_pi=np.array([278., 722., 273., 34.497] + [0.]*25 +
+                          [13.0975, 547.996]),
+             iirf_max=97.0,
+             restart_in=False,
+             restart_out=False,
+             useMultigas=False,
+             tcr_dbl=70.0,
+             F_volcanic=0.0,
+             F_solar=0.0,
+             aviNOx_frac=0.,
+             fossilCH4_frac=0.,
+             natural=np.array([202., 10.]),
+             useStevens=False,
+             efficacy=np.array([1.]*13),
+             scale=np.array([1.]*13),
+             oxCH4_frac=0.61,
+             lifetimes=False,
+             ghg_forcing="Etminan"):
+
+  # Conversion between ppm CO2 and GtC emissions
+  ppm_gtc   = M_ATMOS/1e18*molwt.C/molwt.AIR
+
+  # Conversion between ppb/ppt concentrations and Mt/kt emissions
+  # in the RCP databases ppb = Mt and ppt = kt so factor always 1e18
+  emis2conc = M_ATMOS/1e18*np.asarray(molwt.aslist)/molwt.AIR
+
+  # Funny units for nitrogen emissions - N2O is expressed in N2 equivalent
```

Comment:
I had a similar conversation with Malte, apparently it's an agricultural thing...